mage/query-modules/python/gnn-link-prediction.md

```diff
@@ -0,0 +1,375 @@
+
+#### **Self-loops**
+
+`Self-loop edge` is added to every node to improve link_prediction performance if specified by the user. `Self-loop edges` are added only as edge_type `self`, not in any other way, and to enable this, a custom module has been added. 
+
+#### **Batch training**
+
+In heterogeneous graphs, all edges are used for creating the node’s neighbourhood but trained on only one edge type that can be set by the user. 
+
+> For each gradient descent step, we select a mini-batch of nodes whose final representations at the L-th layer are to be computed. We then take all or some of their neighbours at the L−1 layer. This process continues until we reach the input. This iterative process builds the dependency graph starting from the output and working backwards to the input, as the figure below shows:
+
+<img src={require('../../data/query-modules/python/gnn-link-prediction/gnn-link-prediction-neighborhood-sampling.png').default}/>
+
+> 
+> -- DGL docs
+
+The reader is encouraged to see [DGL mini-batch explanation](https://docs.dgl.ai/guide/minibatch.html) for more details.
+
+## Procedures
+
+### **Set training parameters**
+
+The link prediction module is organized as a stateful module in which the user can run several methods one after another without losing the context. The user should start with setting the parameters that are going to be used in the training. If the graph is **heterogeneous**(more than one **edge type**), `target_relation` parameter must be set so the model could distinguish **supervision edges**(edges used in prediction) from **message passing edges**(used for message aggregation). In the case of **homogeneous graph**, `target relation` will be automatically inferred. `Node_features_property` must also be sent by the user to specify where are saved original node features. Those are needed by **Graph neural networks** to compute **node embeddings**. All other parameters are optional. Here is the description of all parameters supported by **link prediction**:
+
+
+#### **Input**:
+
+| Name            | Type  | Default     | Description |
+| --------------- | ----- | ----------- | ----------- |
+| `hidden_features_size` | mgp.List[int] | `[16, 16]` | Defines the size of each hidden layer in the architecture. Input feature size is determined automatically while converting the original graph to the DGL-compatible one. |
+| `layer_type`    | str   | `graph_attn` | Supported values are `graph_sage` and `graph_attn`. |
+| `num_epochs`    | int   | `100`  | number of epochs for model training.   |
+| `optimizer`     | str   | `ADAM` |  Supported values are `ADAM` and `SGD`. |
+| `learning_rate` | float | `0.01` | optimizer's learning rate.             |
+| `split_ratio`   | float | `0.8`  | split ratio between training and validation set. There is no test dataset because it is assumed that the user first needs to create new edges in the original dataset to test a model on them. |
```

Comment:
```suggestion
| `split_ratio`   | float | `0.8`  | The split ratio between the training and the validation set. There is no test dataset because it's assumed that the user first needs to create new edges in the original dataset to test a model on them. |
```